thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/alert/content/EntityGroupKeyContentFormatter.java

```diff
@@ -93,29 +110,51 @@ protected void updateTemplateDataByAnomalyResults(Map<String, Object> templateDa
     for (String entityName : entityToAnomaliesMap.keySet()) {
       List<AnomalyReportEntity> groupedAnomalies = (List<AnomalyReportEntity>) entityToAnomaliesMap.get(entityName);
       groupedAnomalies.sort((u1, u2) -> anomalyToGroupScoreMap.get(u2).compareTo(anomalyToGroupScoreMap.get(u1)));
-      entityToSortedAnomaliesMap.put(entityName, groupedAnomalies);
+      entityToSortedAnomaliesMap.putAll(entityName, groupedAnomalies);
     }
 
+    // Insert anomaly snapshot image
+    if (whitelistMetricToAnomaliesMap.size() == 1 && whitelistMetricToAnomaliesMap.values().size() == 1) {
```

Comment:
If we only show it when the map size and value size is 1, can we use a Map instead of a multimap?

Reply:
Actually, this condition here is to display the graph when we have only one anomaly. Otherwise, we will list all the anomalies. This map is passed below in line no. 127 which is then processed in the freemarker template file.

Reply:
Make sense!